docs/architecture.md

```diff
@@ -0,0 +1,35 @@
+# OpenShift Hive Architecture
+
+OpenShift Hive is a Kubernetes operator built to provide a declarative API to provision, configure, reshape, and deprovision OpenShift clusters at scale. Hive's API is a set of Kubernetes Custom Resource Definitions, served by the Kubernetes apiserver.
+
+Hive leverages the [OpenShift Installer](https://github.com/openshift/installer) to perform cluster provisioning and deprovisioning.
+
+
+## Components
+
+ 1. **Hive Operator**: The first Hive pod to run is the hive-operator which is essentially a deployer. It manages the Kubernetes Deployments for other components to ensure they are properly configured and running. It also handles migrations and cleanup and anything else that might require programatic logic in a long term Hive deployment. Hive's operator reconciles on a HiveConfig CRD which provides some configuration options for Hive itself.
```

Comment:
s/The first Hive pod to run is the hive-operator which is essentially a deployer/ It is the first pod to run and acts as a deployer`.

Reply:
Rephrased all of these to be a little more brief.